File: C++/Largest-Number.cpp
class Solution {
public:
	static bool cmp(string str1, string str2) {
		return str1 + str2 > str2 + str1;
	}
    string largestNumber(vector<int>& nums) {
        vector<string> res;
        for (int i = 0; i < nums.size(); i ++) {
        	res.push_back(to_string(nums[i]));
        }
        sort(res.begin(), res.end(), cmp);
        string str;
        for(int i = 0; i < nums.size(); i ++) {
        	str += res[i];
        }
        int index = 0;
        while (index < str.size() && str[index] == '0') index ++;
        if (index == str.size()) return "0";
        return str.substr(index);
    }
};

File: Python/Largest-Number.py
class Solution:
    # @param num, a list of integers
    # @return a string
    def cmp1(self, x, y):
        if x + y > y + x:
            return 1
        else:
            return -1
    def largestNumber(self, num):
        seq = []
        for item in num:
            seq.append(str(item))
        seq.sort(self.cmp1)
        return str(int(''.join(seq)))